pkg/reconciler/trigger/resources/dispatcher_test.go

```diff
@@ -257,78 +191,36 @@ func TestMakeDLXDispatcherDeployment(t *testing.T) {
 		BrokerUrlSecretKey: brokerURLKey,
 		BrokerIngressURL:   ingressURL,
 		Subscriber:         sURL,
-		DLX:                true,
 	}
+	for _, o := range opts {
+		o(args)
+	}
+	return args
+}
 
-	got := MakeDispatcherDeployment(args)
-	one := int32(1)
-	want := &appsv1.Deployment{
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: args.Trigger.Namespace,
-			Name:      "testtrigger-dlx-dispatcher",
-			OwnerReferences: []metav1.OwnerReference{{
-				APIVersion:         "eventing.knative.dev/v1",
-				Kind:               "Trigger",
-				Name:               triggerName,
-				Controller:         &TrueValue,
-				BlockOwnerDeletion: &TrueValue,
-			}},
-			Labels: map[string]string{
-				"eventing.knative.dev/broker":     brokerName,
-				"eventing.knative.dev/brokerRole": "dispatcher",
-			},
-		},
-		Spec: appsv1.DeploymentSpec{
-			Replicas: &one,
-			Selector: &metav1.LabelSelector{
-				MatchLabels: map[string]string{
-					"eventing.knative.dev/broker":     brokerName,
-					"eventing.knative.dev/brokerRole": "dispatcher",
-				},
-			},
-			Template: corev1.PodTemplateSpec{
-				ObjectMeta: metav1.ObjectMeta{
-					Labels: map[string]string{
-						"eventing.knative.dev/broker":     brokerName,
-						"eventing.knative.dev/brokerRole": "dispatcher",
-					},
-				},
-				Spec: corev1.PodSpec{
-					Containers: []corev1.Container{{
-						Name:  "dispatcher",
-						Image: image,
-						Env: []corev1.EnvVar{{
-							Name:  system.NamespaceEnvKey,
-							Value: system.Namespace(),
-						}, {
-							Name: "RABBIT_URL",
-							ValueFrom: &corev1.EnvVarSource{
-								SecretKeyRef: &corev1.SecretKeySelector{
-									LocalObjectReference: corev1.LocalObjectReference{
-										Name: secretName,
-									},
-									Key: brokerURLKey,
-								},
-							},
-						}, {
-							Name:  "QUEUE_NAME",
-							Value: queueName,
-						}, {
-							Name:  "SUBSCRIBER",
-							Value: subscriberURL,
-						}, {
-							Name:  "BROKER_INGRESS_URL",
-							Value: brokerIngressURL,
-						}, {
-							Name:  "REQUEUE",
-							Value: "false",
-						}},
-					}},
-				},
-			},
-		},
+func withDelivery(delivery *eventingduckv1.DeliverySpec) func(*DispatcherArgs) {
+	return func(args *DispatcherArgs) {
+		args.Delivery = delivery
 	}
-	if diff := cmp.Diff(want, got); diff != "" {
-		t.Error("unexpected diff (-want, +got) = ", diff)
+}
+
+func withDLX(args *DispatcherArgs) {
+	args.DLX = true
+}
+
+func withPrefetch(c string) func(*DispatcherArgs) {
+	return func(args *DispatcherArgs) {
+		if args.Trigger.ObjectMeta.Annotations == nil {
+			args.Trigger.ObjectMeta.Annotations = map[string]string{prefetchAnnotation: c}
+		} else {
+			args.Trigger.ObjectMeta.Annotations[prefetchAnnotation] = c
+		}
 	}
 }
+
+func Int32Ptr(i int32) *int32 {
+	return &i
+}
+func StringPtr(s string) *string {
```

Comment:
anything planned for this stringPtr function? is not been used

Reply:
removed, thanks!